src/integrations/prefect-databricks/prefect_databricks/credentials.py

```diff
@@ -12,51 +14,227 @@ class DatabricksCredentials(Block):
+                "`client_secret` must be provided."
+            )
+
+        if not has_token and not has_all_sp:
+            raise ValueError(
+                "Must provide either `token` for PAT authentication or "
+                "`client_id` and `client_secret` for service principal authentication."
+            )
+
+        return values
+
+    def _get_oauth_token(self) -> str:
+        """
+        Acquires an OAuth token using the service principal credentials.
+
+        Implements token caching and automatic refresh when the token is expired
+        or about to expire (within 60 seconds of expiration).
+
+        For Azure Databricks with a tenant_id, uses Microsoft Entra ID (Azure AD)
+        token endpoint. For Databricks-managed service principals (no tenant_id),
+        uses the Databricks OIDC endpoint.
+
+        Returns:
+            A valid OAuth access token.
+
+        Raises:
+            RuntimeError: If token acquisition fails.
+        """
+        if (
+            self._cached_token is not None
+            and self._token_expiry is not None
+            and time.time() < self._token_expiry - 60
+        ):
+            return self._cached_token
+
+        if self.tenant_id is not None:
+            # Azure AD (Microsoft Entra ID) authentication
+            token_url = (
+                f"https://login.microsoftonline.com/{self.tenant_id}/oauth2/v2.0/token"
+            )
+            headers = {
+                "Content-Type": "application/x-www-form-urlencoded",
+            }
+            # Azure Databricks resource ID
+            azure_databricks_resource_id = "2ff814a6-3304-4ab8-85cb-cd0e6f879c1d"
+            data = {
+                "grant_type": "client_credentials",
+                "client_id": self.client_id,
+                "client_secret": self.client_secret.get_secret_value(),
+                "scope": f"{azure_databricks_resource_id}/.default",
+            }
+        else:
+            # Databricks-managed service principal authentication
+            token_url = f"https://{self.databricks_instance}/oidc/v1/token"
+            credentials = f"{self.client_id}:{self.client_secret.get_secret_value()}"
+            encoded_credentials = base64.b64encode(credentials.encode()).decode()
+            headers = {
+                "Authorization": f"Basic {encoded_credentials}",
+                "Content-Type": "application/x-www-form-urlencoded",
+            }
+            data = {
+                "grant_type": "client_credentials",
+                "scope": "all-apis",
+            }
+
+        with Client() as client:
+            response = client.post(token_url, headers=headers, data=data)
+
+            if response.status_code != 200:
+                raise RuntimeError(
+                    f"Failed to acquire OAuth token: {response.status_code} - {response.text}"
+                )
+
+            token_data = response.json()
+            self._cached_token = token_data["access_token"]
```

Comment:
Handle the edge case where `access_token` isn't in the response and raise a user-friendly exception.